File: src/rust/src/lib/data/synth.rs
use crate::lib::data::midi_command::MIDICommand;

#[derive(Clone)]
pub struct Synth {
    pub name: String,
    pub id: String,
    pub manufacturer: String,
    pub commands: Vec<MIDICommand>,
}

impl Synth {
    pub fn new(id: String) -> Self {
        Self {
            name: String::new(),
            id: id,
            manufacturer: String::new(),
            commands: Vec::new(),
        }
    }
}

File: src/cpp/midi_prog/synth.cpp
#include "synth.h"
#include "midicommand.h"

Synth::Synth(const Synth &s)
{
    this->name = s.name;
    this->id = s.id;
    this->manufacturer = s.manufacturer;
    this->commands = s.commands;
}

Synth::Synth(std::string id) : id(id)
{

}


void Synth::setName(std::string name)
{
    this->name = name;
}

void Synth::setManufacturer(std::string manufacturer)
{
    this->manufacturer = manufacturer;
}

std::vector<MIDICommand> *Synth::getCommands()
{
    return &this->commands;
}

std::string Synth::getId()
{
    return this->id;
}
